pkg/service/repair/intensity_handler.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2024 ScyllaDB
+
+package repair
+
+import (
+	"context"
+	"math"
+
+	"github.com/scylladb/go-log"
+	"github.com/scylladb/scylla-manager/v3/pkg/util/uuid"
+	"go.uber.org/atomic"
+)
+
+type sizeSetter interface {
+	SetSize(size int)
+}
+
+type intensityHandler struct {
+	taskID           uuid.UUID
+	runID            uuid.UUID
+	logger           log.Logger
+	maxHostIntensity map[string]Intensity
+	intensity        *atomic.Int64
+	maxParallel      int
+	parallel         *atomic.Int64
+	poolController   sizeSetter
+}
+
+const (
+	maxIntensity     Intensity = 0
+	defaultIntensity Intensity = 1
+	defaultParallel            = 0
+	chanSize                   = 10000
+)
+
+// SetIntensity sets the value of '--intensity' flag.
```

Comment:
nit: I think it doesn't set the value of `--intensity` flag, but rather updates ongoing repairs with given value for intensity (link https://manager.docs.scylladb.com/stable/sctool/repair.html#intensity)